test/server/graphql/v2/mutation/MemberInvitationMutations.test.ts

```diff
@@ -0,0 +1,437 @@
+import * as utils from '../../../../utils';
+
+let collectiveAdminUser, collective;
+let sandbox, sendEmailSpy;
+
+describe('MemberInvitationMutations', () => {
+  before(async () => {
+    await utils.resetTestDB();
+    sandbox = createSandbox();
+    sendEmailSpy = sandbox.spy(emailLib, 'sendMessage');
+    collectiveAdminUser = await fakeUser();
+    collective = await fakeCollective({ name: 'webpack test collective', admin: collectiveAdminUser });
+  });
+  after(() => sandbox.restore());
+
+  const inviteMemberMutation = gqlV2/* GraphQL */ `
+    mutation InviteMember(
+      $memberAccount: AccountReferenceInput!
+      $account: AccountReferenceInput!
+      $role: MemberRole!
+      $description: String
+      $since: DateTime
+    ) {
+      inviteMember(
+        memberAccount: $memberAccount
+        account: $account
+        role: $role
+        description: $description
+        since: $since
+      ) {
+        id
+        role
+        description
+        since
+      }
+    }
+  `;
+
+  describe('inviteMember', () => {
+    it('should create a new member invitation and its related activity', async () => {
+      sendEmailSpy.resetHistory();
+      const randomUserToInvite = await fakeUser();
+      const result = await utils.graphqlQueryV2(
+        inviteMemberMutation,
+        {
+          memberAccount: { id: idEncode(randomUserToInvite.collective.id, IDENTIFIER_TYPES.ACCOUNT) },
+          account: { id: idEncode(collective.id, IDENTIFIER_TYPES.ACCOUNT) },
+          description: 'new user 2 as MEMBER',
+          role: roles.MEMBER,
+          since: new Date('01 January 2022').toISOString(),
+        },
+        collectiveAdminUser,
+      );
+
+      const activity = await models.Activity.findAll({
+        where: {
+          type: ActivityTypes.COLLECTIVE_CORE_MEMBER_INVITED,
+          CollectiveId: collective.id,
+        },
+      });
+      const createdMemberInvitation = result.data.inviteMember;
+
+      expect(result.errors).to.not.exist;
+      expect(activity.length).to.equal(1);
+      expect(activity[0].data.memberCollective.id).to.equal(randomUserToInvite.collective.id);
+      expect(createdMemberInvitation.role).to.equal(roles.MEMBER);
+      expect(createdMemberInvitation.description).to.equal('new user 2 as MEMBER');
+      expect(createdMemberInvitation.since.toISOString()).to.equal(new Date('01 January 2022').toISOString());
+      expect(sendEmailSpy.callCount).to.equal(1);
+      expect(sendEmailSpy.args[0][1]).to.equal('Invitation to join webpack test collective on Open Collective');
+    });
+    it('must be authenticated as an admin of the collective', async () => {
+      sendEmailSpy.resetHistory();
+      const randomUser = await fakeUser();
+      const anotherRandomUser = await fakeUser();
+      const result = await utils.graphqlQueryV2(
+        inviteMemberMutation,
+        {
+          memberAccount: { id: idEncode(anotherRandomUser.id, IDENTIFIER_TYPES.ACCOUNT) },
```

Comment:
Actually related to https://github.com/opencollective/opencollective-api/pull/7616/files#r893101351, this should be: 

```suggestion
          memberAccount: { id: idEncode(anotherRandomUser.CollectiveId, IDENTIFIER_TYPES.ACCOUNT) },
```

Or

```suggestion
          memberAccount: { id: idEncode(anotherRandomUser.collective.id, IDENTIFIER_TYPES.ACCOUNT) },
```

Same for the tests below. I'm including that in my upcoming commit, let me know if it doesn't make sense to you

Reply:
It makes sense now. I got confused with the two tables `Users` and `Collective` (with type User).

Reply:
`Collectives` is a confusing table name that should really be named `Accounts` or `Profiles`. Because of its wide usage, it's difficult to rename, but we'll get there someday!